src/Slider/Slider.tsx

```diff
@@ -101,19 +128,10 @@ const Slider: React.FunctionComponent<SliderProps> = (props: SliderProps): React
                     disabled={props.disabled}
                 />
                 <div className={"custom-slider-holder" + (props.theme ? ` ${props.theme}` : " primary")}>
-                    <div className="custom-slider-track">
-                        <div
-                            className="custom-slider-slider-before"
-                            style={{ width: getPercentage(props.value) + "%" }}
-                        />
-                        <div
-                            className="custom-slider-slider-after"
-                            style={{ width: (100 - getPercentage(props.value)) + "%" }}
-                        />
-                        <div
-                            className="custom-slider-thumb"
-                            style={{ left: getPercentage(props.value) + "%" }}
-                        >
+                    <div className={"custom-slider-track" + (size / props.step <= 30 ? " with-transitions" : "")}>
```

Comment:
All this logic seems to be very custom and advanced to be hardcoded inline. Either extract it to a function with a descriptive name or try to make it more readable somehow.

Reply:
moved the logic to a method